### ISSUE TITLE
feature request: support  __webpack_public_path__ for set publicPath dynamically at runtime 

### ISSUE THREAD

**Issue by Zhangdroid** [faceyspacey/extract-css-chunks-webpack-plugin#51]
hi~

I'm using your great work react-universal-component with extract-css-chunks-webpack-plugin and have some troubles.

We have three different test environments: dev, oa, and pre, and they have different CDN like https://devwww.examplecdn.com, https://oawww.examplecdn.com. The static files only compiled on dev environment and copy to oa ,pre and online site, so we want set publicPath(CDN) dynamically at runtime.

Webpack [document](https://webpack.js.org/concepts/output/#advanced) said we can use `__webpack_public_path__` to do this like:

```js
// publicPath.js
// On server side configs will read process.env.ENV_TYPE and return configs for this environments
// On client side configs will read window.ENV_TYPE
import configs from 'configs';

__webpack_public_path__ = `${configs.CDN_URL}/`;
```

On server side , we manual add public path(it would be nice if webpack-flush-chunks can provide the public path option):

```js
import configs from 'configs';

export default ({
  content, scripts
}) => `
<!doctype html>
<html lang="en">
<head>
</head>
<body>
  <div id="root">${content}</div>
  ...
  <script type="text/javascript">
    window.ENV_TYPE="${process.env.ENV_TYPE}";
  </script>
  ${scripts.map(name => `<script type="text/javascript" src="//${configs.CDN_URL}/${name}"></script>`)}
</body>
</html>
`;

```

On client side, the javascript files works fine when I using `__webpack_public_path__`, but css files not.

I saw extract-text-webpack-plugin have same [problem](https://github.com/webpack-contrib/extract-text-webpack-plugin/issues/513) but no one answer it yet. Is there anyone can help for this?

**Comment by faceyspacey** [comment]
I'm not exactly sure what to do about that, @kmcclosk might know??

**Comment by Zhangdroid** [comment]
@faceyspacey I'm not familiar with how webpack works, but  I just figured out how CSS files dynamically load on client side, and there are two simple ways to make this work:

1. webpack-flush-chunks add the `publicPath` option that user can dynamically set on server side, rewrite `publicPath` that stats.json provided, then `window.__CSS_CHUNKS__` will have the right path.

2.  user provide a global variable like `__PUBLIC_PATH__`, then babel-plugin-universal-import add it before chunk path.

```js
// babel-plugin-universal-import/importCss.js
...
function getHref(chunkName) {
  if (typeof window === 'undefined' || !window.__CSS_CHUNKS__) return null
  return window.__PUBLIC_PATH__ ?  window.__PUBLIC_PATH__ : '' + window.__CSS_CHUNKS__[chunkName]
}
```
If those ways are ok, I can make a PR for this.

**Comment by faceyspacey** [comment]
#1 sounds about right. Go for it.

**Comment by Zhangdroid** [comment]
@faceyspacey I just found another way: we can read `__webpack_public_path__` in webpack-flush-chunks and babel-plugin-universal-import. Since `__webpack_public_path__` [equals the config options output.publicPath](https://webpack.js.org/api/module-variables/#__webpack_public_path__-webpack-specific-), we can use `__webpack_public_path__` as publicPath instead of publicPath of stats.json(only need user set `__webpack_public_path__` on both client entry and server entry).

Also, I think isn't it more reasonable that `window.__CSS_CHUNKS__` just export file name of css chunks and add public path with babel-plugin-universal-import?
```js
// webpack-flush-chunks/createApiWithCss.js
export const createCssHash = ({
  assetsByChunkName,
  publicPath
}: {
  assetsByChunkName: FilesMap,
  publicPath: string
}): CssChunksHash =>
  Object.keys(assetsByChunkName).reduce((hash, name) => {
    ...
    if (file) hash[name] = `${file}`
    return hash
  }, {})
```

```js
// babel-plugin-universal-import/importCss.js
...
function getHref(chunkName) {
  if (typeof window === 'undefined' || !window.__CSS_CHUNKS__) return null
  return  __webpack_public_path__ + window.__CSS_CHUNKS__[chunkName]
}
```

**Comment by faceyspacey** [comment]
I don't have much time to focus on this right now, so a change that might be just fine, but could potentially break things for lots of existing users, makes me nervous without being able to analyze it myself. 

Also ideally we don't touch the babel plugin.

**Comment by zackljackson** [comment]
Will be resolved by #71
